examples/http_server/module/src/lib.rs

```diff
@@ -80,28 +181,166 @@ impl CommandHandler for StaticHttpHandler {
                 .header(http::header::CONTENT_TYPE, "text/html; charset=UTF-8")
                 .body(include_bytes!("../static/index.html").to_vec())
                 .context("Could not build response"),
-            "/test_google" => {
-                // create a public HTTP client pseudo-node
-                let http_client = oak::http::client::init("")
-                    .context("Could not create HttpClient pseudo-node")?;
-                let request = http::Request::builder()
-                    .method(http::Method::GET)
-                    .uri("http://www.google.com")
-                    .body(vec![])
-                    .context("Could not build request")?;
-                http_client
-                    .send_request(request, &Label::public_untrusted())
-                    .context("Could not get response")
+            _ => http::response::Builder::new()
+                .status(http::StatusCode::NOT_FOUND)
+                .body("not found".to_string().into_bytes())
+                .context("Could not build response"),
+        };
+
+        let response = match response {
+            Ok(rsp) => rsp,
+            Err(err) => http::response::Builder::new()
+                .status(http::StatusCode::INTERNAL_SERVER_ERROR)
+                .body(err.to_string().into_bytes())
+                .context("Could not build response")?,
+        };
+
+        info!("Sending the response on the invocation channel!");
+        let res = invocation.send(&response);
+        invocation.close_channels();
+        res?;
+        Ok(())
+    }
+}
+
+/// A handler corresponding to an HTTP server that redirects every request sent to `/redirect` to
+/// `https://localhost:8080` and responds with `OK` (200) if the redirect is successful, or
+/// responds with `NOT_FOUND` (400) to any other request.
+pub struct RedirectHandler {
+    invocation_sender: Sender<RedirectInvocation>,
+}
+
+impl oak::WithInit for RedirectHandler {
+    type Init = RedirectHandlerInit;
+
+    fn create(init: Self::Init) -> Self {
+        oak::logger::init(init.log_sender.unwrap(), log::Level::Debug).unwrap();
+        RedirectHandler {
+            invocation_sender: init.invocation_sender.unwrap(),
+        }
+    }
+}
+
+impl CommandHandler for RedirectHandler {
+    type Command = Invocation;
+
+    fn handle_command(&mut self, invocation: Invocation) -> anyhow::Result<()> {
+        let request = invocation.clone().receiver.unwrap().receive()?;
```

Comment:
I thought the intention was that the client will send a request with a TLS endpoint label. At the moment, the redirect handler is public. If the client sends a redirect request with a TLS endpoint label, the redirect handler will not be able to receive the request on this channel and it will fail here.

I think the public handler that receives the invocation should split the invocation. It should send the request receiver channel handle to a node with the right TLS endpoint label (where the request inside the channel can be received)m together with the new invocation channel for the HTTP client node request.

The public node should respond immediately with "success" on the public response channel, otherwise the client will hang as it will never receive a response. The public node will never know whether the redirect request was successful, so can just always return success.

Reply:
I have pushed a change that splits the invocation (if I understood correctly). 

For a non-public request label to work, I think we have to use the router pattern. Basically, the `StaticHttpHandler` should create a separate node for each request label. Is that right? Did not we earlier say that only the planner/Main node should create new nodes?

Reply:
I don't hink we need a full router pattern. The router pattern usually creates one node per label. As all requests will have the same label, one node with the non-public label should be enough.

But there should still be a public node that works a bit like a router node, without creating a new node for every request:

- It should receive the invocations from the HTTP server
- it should create 2 channels for the http client node invocation
- it should send the invocation to the http client node
- It should send the following to the non-public node
  - the request reader channel from the server invocation
  - the other sides of the channels for the invocation that was sent to the http client node (request writer and response reader)
- it should respond with a success message on the response writer it received in the server invocation and close this response writer

The non-public node should then:

- Read the request from the server invocation using the request receiver it received
- Translate that request into a request that it writes on the request writer for the client invocation
- Wait for a response from the client using the response reader from the client invocation
- Read the response (and ignore it)
- Close the response reader

Nodes can only be created by public nodes at the moment, so either the initial node can create all the nodes and channels that are needed, or it can create a dedicated planner node with a public label which then creates the other nodes and all the channels used for the invocations.